### DISCUSSION TITLE
How do I open and view a mysqldump-generated .sql file in DBeaver?

### DISCUSSION THREAD

**Discussion by tromlet** [dbeaver/dbeaver#20191] [General]
Hey all -

I'm doing some work on an exported database from an old server that we're shutting down, and I was wondering how I open a `.sql` file that I generated from that server with mysqldump? Server was running MySQL 5.1.73 (specifically: `mysql  Ver 14.14 Distrib 5.1.73, for redhat-linux-gnu (i386) using readline 5.1`).

The SQLite connection doesn't work, and both MariaDB and MySQL require a server for me to connect to, I can't just pick up a file. :/

**Comment by LonwoLonwo** [comment]
Hello @tromlet 

You can open any script in DBeaver.

![2023-06-05 10_30_00-](https://github.com/dbeaver/dbeaver/assets/45152336/b1ed837f-cad1-4a32-a381-d33aabec2798)

And yes, it will not be bounded with any connection this way. In this case, you can read this file and change. But not execute queries, before you will have working connection to MySQL or MariaDB 

![2023-06-05 10_29_47-Window](https://github.com/dbeaver/dbeaver/assets/45152336/bbde76a9-2dd3-49f2-a754-a1e03a0a4cb6)

**Comment by tromlet** [comment]
Interesting. I DID try that, but my DBeaver is crashing every time I open that `.sql` file. :(

It's an older version, as I've run out of space on my desktop and I need to reinstall my OS on a newer, bigger, SSD, so I guess I'll have to wait on that.

If I'm reading you correctly: I **CANNOT** run SQL statements against a file to pull data from it?

**Reply by LonwoLonwo** [reply]
What kind of statements do you want to run without connection?

And how big is your SQL file?

**Reply by tromlet** [reply]
Some basic select statements - I don't need any timestamping or any other data that Drupal might put into a user registration, but I DO want the user's name, email address, company name, and potentially other data, exported to CSV, so that I can send it off to one of my sales guys.

It's about 60 MB in size.

**Reply by LonwoLonwo** [reply]
You can use a CSV connection if you have a CSV file with data. Only selects are working there.

![2023-06-07 20_02_04-Connect to a database](https://github.com/dbeaver/dbeaver/assets/45152336/6ad22244-7fb0-4de1-8baf-064bd1d654d5)
